File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Relics/PaperCrane.ts
import FileHelper from "../FileHelper";
import { LocalizationProperty } from "../Gen/DB/Localization";
import AbstractRelic from "./AbstractRelic";

export class PaperCrane extends AbstractRelic{
    public static readonly Id = "PaperCrane";
    Desc: string = LocalizationProperty.Read("纸鹤-效果描述");
    Flavor: string = LocalizationProperty.Read("纸鹤-特殊描述");
    Icon: string = FileHelper.FormatPath("Relics/paperCrane.png");
    Id: string = "PaperCrane";
    Name: string = LocalizationProperty.Read("纸鹤");
    
    OnLose() {
    }

    OnObtain() {
    }
    
}

File: Unity/Assets/Scripts/ModTheSpire/Relics/PaperCrane.cs
namespace ModTheSpire.Relics
{
    //TODO 未实现的遗物
    public class PaperCrane : AbstractRelic
    {
        public override string Id { get; }
        public override string Name { get; }
        public override string Icon { get; }
        public override string Desc { get; }
        public override string Flavor { get; }
        public override void OnObtain()
        {
            throw new System.NotImplementedException();
        }

        public override void OnLose()
        {
            throw new System.NotImplementedException();
        }
    }
}